### PR TITLE
TEAMNADO-4773 Refactor Activation Keys Overview request to merge to "master"

### PR DESCRIPTION
**Pull request by BrooklynLCodes**
# Description

Associated Jira ticket: # (issue)

Please include a summary of the change, what this fixes/creates/improves.
These change include the removal of the unnecessary button wrappers, and added logic to justify error handling in the additional repositories card. These changes were made to address the refactor comments in Jira Link for plausible, justified code clean up.
# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Comment by jmbuscemi** [comment]
@BrooklynLCodes reminder to put the jira ticket and its title as the title of the PR to help identifying this work for those reviewing and in the git history.

**Comment by BrooklynLCodes** [comment]
> @BrooklynLCodes reminder to put the jira ticket and its title as the title of the PR to help identifying this work for those reviewing and in the git history.
Absolutely, will modify.

File: src/Components/ActivationKey/AdditionalRepositoriesCard.js [hunk 1]
```diff
@@ -34,45 +36,48 @@ const AdditionalRepositoriesCard = (props) => {
...
+              onClick={handleEditAdditionalRepositoriesToggle}
+            />
+            <AddAdditionalRepositoriesModal
+              title="Additional Repositories"
+              isOpen={isEditAdditionalRepositoriesModalOpen}
+              handleModalToggle={handleEditAdditionalRepositoriesToggle}
+              modalSize="large"
+              repositories={availableRepositories}
+              isLoading={isLoading}
+              error={error}
```
Comment (lukegrn) [inline src/Components/ActivationKey/AdditionalRepositoriesCard.js]: `title` and `modalSize` are no longer passed to this component. See #176 where that was removed.

File: src/Components/ActivationKey/AdditionalRepositoriesCard.js [hunk 2]
```diff
@@ -34,45 +36,48 @@ const AdditionalRepositoriesCard = (props) => {
...
-        <AdditionalRepositoriesTable
-          repositories={activationKey.additionalRepositories}
-        />
-      </CardBody>
-    </Card>
-  );
+  if (isLoading && !error) {
+    return <Loading />;
+  } else if (!isLoading && !error) {
+    return AdditionalReposCard();
```
Comment (lukegrn) [inline src/Components/ActivationKey/AdditionalRepositoriesCard.js]:
```suggestion
    return <AdditionalReposCard />;
```

File: src/Components/ActivationKey/__tests__/ActivationKey.test.js [hunk 3]
```diff
@@ -108,11 +108,9 @@ describe('ActivationKey', () => {
...
 
   it('renders correctly', async () => {
-    render(<PageContainer />);
+    const { container } = render(<PageContainer />);
     await waitFor(() => expect(useUser).toHaveBeenCalledTimes(1));
-    expect(screen.getByText('System Purpose')).toBeInTheDocument();
-    expect(screen.getByText('Workload')).toBeInTheDocument();
-    expect(screen.getByText('Additional repositories')).toBeInTheDocument();
+    expect(container).toMatchSnapshot();
   });
```
Comment (lukegrn) [inline src/Components/ActivationKey/__tests__/ActivationKey.test.js]: We shouldn't have to switch this test to a snapshot test - it makes me nervous to see a test change when functionality shouldn't. That makes me think the functionality _is_ changing.

File: src/Components/ActivationKey/AdditionalRepositoriesCard.js [hunk 4]
```diff
@@ -57,18 +49,19 @@ const AdditionalRepositoriesCard = (props) => {
...
             handleModalToggle={handleEditAdditionalRepositoriesToggle}
-            keyName={activationKey.name}
             repositories={availableRepositories}
             isLoading={isLoading}
             error={error}
           />
         </TextContent>
         <AdditionalRepositoriesTable
-          repositories={activationKey.additionalRepositories}
+          repositories={activationKey.availableRepositories}
```
Comment (lukegrn) [inline src/Components/ActivationKey/AdditionalRepositoriesCard.js]:
```suggestion
          repositories={activationKey.additionalRepositories}
```
This piece of data needs to stay what it was

File: src/Components/AdditionalRepositoriesTable/AdditionalRepositoriesTable.js [hunk 5]
```diff
@@ -126,7 +126,7 @@ const AdditionalRepositoriesTable = (props) => {
           })}
         </Tbody>
       </TableComposable>
-      {repositories.length === 0 && <NoAdditionalRepositories />}
+      {repositories?.length === 0 && <NoAdditionalRepositories />}
```
Comment (lukegrn) [inline src/Components/AdditionalRepositoriesTable/AdditionalRepositoriesTable.js]: Why do we need to make this null safe? Even when there are no additional repositories the response comes back with an empty array. Just the length check should suffice

File: src/Components/ActivationKey/AdditionalRepositoriesCard.js [hunk 6]
```diff
@@ -57,11 +49,12 @@ const AdditionalRepositoriesCard = (props) => {
...
             explicitly added to the activation key.
           </Text>
-          <ButtonWrapper />
+          <AddAdditionalRepositoriesButton
+            onClick={handleEditAdditionalRepositoriesToggle}
+          />
           <AddAdditionalRepositoriesModal
             isOpen={isEditAdditionalRepositoriesModalOpen}
             handleModalToggle={handleEditAdditionalRepositoriesToggle}
-            keyName={activationKey.name}
```
Comment (lukegrn) [inline src/Components/ActivationKey/AdditionalRepositoriesCard.js]: The keyName must be passed in - without it you cannot save additional repositories.

**Review by lukegrn — APPROVED** [review]

**Pull request merged**